### PR TITLE
RHIROS-1102 Kruize - Experiment creation failure

### PR DESCRIPTION
**Pull request by saltgen**

### CODE REVIEW

File: internal/utils/kruize/kruize_api.go [hunk 1]
```diff
@@ -45,14 +45,36 @@ func Create_kruize_experiments(experiment_name string, k8s_object []map[string]i
...
+		log.Error("Performance profile does not exist")
+		log.Info("Tring to create resource_optimization_openshift performance profile")
+		utils.Setup_kruize_performance_profile()
+	}
+
+	// Re-attempting experiment creation request
+	res, err = http.Post(url, "application/json", bytes.NewBuffer(payload))
+	if err != nil {
+		return nil, fmt.Errorf("error Occured while creating experiment: %v", err)
+	}
```
Comment (patilsuraj767) [inline internal/utils/kruize/kruize_api.go:69]: This will duplicate every create experiment request.

**Review by patilsuraj767 — CHANGES_REQUESTED** [review]

File: internal/utils/kruize/kruize_api.go (hunk 1 above)
Comment (patilsuraj767) [reply internal/utils/kruize/kruize_api.go:69]: Can we check the response of create experiment and if it says `Performance Profile doesn't exist` then create the performance profile and give recursive call to `Create_kruize_experiments` function.

File: internal/utils/kruize/kruize_api.go [hunk 2]
```diff
@@ -53,6 +53,16 @@ func Create_kruize_experiments(experiment_name string, k8s_object []map[string]i
...
 		return nil, fmt.Errorf("can not unmarshal response data: %v", err)
 	}
+
+	// Temporary fix
+	// Currently, once Kruize pod inits it does not load performance-profile from DB
+	if strings.Contains(resdata["message"].(string), "Performance Profile doesn't exist") {
+		log.Error("Performance profile does not exist")
+		log.Info("Tring to create resource_optimization_openshift performance profile")
+		utils.Setup_kruize_performance_profile()
+		_, _ = Create_kruize_experiments(experiment_name, k8s_object)
```
Comment (patilsuraj767) [inline internal/utils/kruize/kruize_api.go:63]: We need to return the response to main calling function. 

```suggestion
		if container_names, err := Create_kruize_experiments(experiment_name, k8s_object); err != nil {
				return nil, err
			} else {
				return container_names, nil
			}
```

**Review by patilsuraj767 — CHANGES_REQUESTED** [review]

File: internal/utils/kruize/kruize_api.go [hunk 3]
```diff
@@ -45,14 +46,25 @@ func Create_kruize_experiments(experiment_name string, k8s_object []map[string]i
 	}
 	res, err := http.Post(url, "application/json", bytes.NewBuffer(payload))
 	if err != nil {
-		return nil, fmt.Errorf("error Occured while creating experiment: %v", err)
+		log.Errorf("error Occured while creating experiment: %v", err)
```
Comment (patilsuraj767) [inline internal/utils/kruize/kruize_api.go:49]: We need to `return` from here if there is any issue while calling kruize API.

Comment (saltgen) [reply internal/utils/kruize/kruize_api.go:49]: Won't this exit the function?

Comment (saltgen) [reply internal/utils/kruize/kruize_api.go:49]: Ah, for other scenarios, you mean

Comment (patilsuraj767) [reply internal/utils/kruize/kruize_api.go:49]: When kruize return "performance profile already exist" at that time this is not triggered. err is only set when there is any issue in making http request. for example - kruize is down.

**Pull request merged**